src/Segmentation/Watersheds/SegmentWithWatershedAndDistanceMap/Code.py

```diff
@@ -0,0 +1,133 @@
+#        depth that is 100 percent of the maximum value in the image.
+#        A level of 0.0 produces the basic segmentation, which will typically be very oversegmented.
+#        Level values of interest are typically low (i.e. less than about 0.40 or 40%),
+#        since higher values quickly start to undersegment the image.
+
+
+import argparse
+
+import itk
+import numpy as np
+
+parser = argparse.ArgumentParser(
+    description="Segment an image using the watershed method and the signed Maurer distance map."
+)
+parser.add_argument("input_image_filename")
+parser.add_argument("reversed_input_output_filename")
+parser.add_argument("distance_map_output_filename")
+parser.add_argument("watershed_output_filename")
+parser.add_argument("segmentation_result_output_filename")
+parser.add_argument("binarizing_radius", type=int)
+parser.add_argument("majority_threshold", type=int)
+parser.add_argument("watershed_threshold", type=float)
+parser.add_argument("level", type=float)
+parser.add_argument("cleaning_structuring_element_radius", type=int)
+args = parser.parse_args()
+
+dimension = 3
+
+uchar_pixel_type = itk.UC
+uchar_image_type = itk.Image[uchar_pixel_type, dimension]
+
+float_pixel_type = itk.F
+float_image_type = itk.Image[float_pixel_type, dimension]
+
+rgb_pixel_type = itk.RGBPixel[uchar_pixel_type]
+RGBImageType = itk.Image[rgb_pixel_type, dimension]
+
+stack_image = itk.imread(args.input_image_filename)
+
+# Create bubble image: get a binarized version of the input image
+index_radius = itk.Size[dimension]()
+index_radius.Fill(args.binarizing_radius)
+
+bubble_image = itk.voting_binary_iterative_hole_filling_image_filter(
+    stack_image,
+    radius=index_radius,
+    background_value=0,
+    foreground_value=255,
+    majority_threshold=args.majority_threshold,
+)
+
+# Write bubble image
+itk.imwrite(bubble_image, args.reversed_input_output_filename)
+
+# Watershed on bubbles
+dimension = len(np.shape(bubble_image))
+
+bbl_cast_image = itk.cast_image_filter(bubble_image, ttype=(uchar_image_type, float_image_type))
+
+# Convert to itk array and normalize
+bubble_image_preclamp = itk.multiply_image_filter(bbl_cast_image, constant=255.0)
+
+bubble_image_clamp = itk.clamp_image_filter(bubble_image_preclamp, bounds=(0, 255),)
+
+# Get the distance map of the input image
+distance_map_image = itk.signed_maurer_distance_map_image_filter(
+    bubble_image_clamp, inside_is_positive=False
+)
+
+itk.imwrite(distance_map_image, args.distance_map_output_filename)
+
+# Apply the watershed segmentation
+watershed_image = itk.watershed_image_filter(
+    distance_map_image, threshold=args.watershed_threshold, level=args.level,
+)
+
+# distance_map_array = itk.array_view_from_image(distance_map_image)
+# watershed_array = itk.array_view_from_image(watershed_image)
+
+# Cast to unsigned char so that it can be written as a TIFF image
+# WatershedImageFilter produces itk.ULL, but CastImageFilter does not wrap
+# itk.ULL: see ITK issue 2551
```

Comment:
Judging by https://github.com/InsightSoftwareConsortium/ITK/blob/v5.2.1/Modules/Filtering/Colormap/wrapping/itkScalarToRGBColormapImageFilter.wrap#L9, and default 64 bit identifier type, and that being ULL on Windows,